src/plugins/utils.tsx

```diff
@@ -78,3 +78,38 @@ export const loadToDom = <PluginOptions extends {}>(
   }
   return element;
 };
+
+export const toggleAttributeOnClick = ({
+  name,
+  attribute,
+}: {
+  name: string;
+  attribute: string;
+}) => {
+  return new Plugin({
+    key: new PluginKey(`update${name}AttributeOnClick`),
+    props: {
+      handleClickOn(view, _pos, _node, _nodePos, event) {
+        if (view.editable) {
+          return false;
+        }
+        const element = event.target as HTMLElement;
+        if (!element.hasAttribute(attribute)) {
+          console.log("target doesn't have html attribute", attribute);
+          return false;
+        }
+        const currentValue = element.getAttribute(attribute);
+        if (!currentValue) {
+          console.log("element attribute has no currentValue");
+          return false;
+        }
+        const newValue = currentValue === "false" ? "true" : "false";
```

Comment:
This can be simplified to `const newValue = String(currentValue === "false")`.

Is there a reason we're using strings here?

Reply:
We're using strings because `getAttribute` only returns strings and `setAttribute` only takes strings, and it didn't seem worth it to me to transform it to a boolean and back for the one line in between.

I feel like `const newValue = String(currentValue === "false")` is considerably less readable than the ternary. Is there a reason it would be better beyond being very slightly shorter?

Reply:
It took me a little bit yesterday to figure out whether there was a special reason you were doing the ternary that way, and to realize that it was equivalent to checking `currentValue === "false`. The second one feels more readable to me, but also we could move the String conversion later. It should even happen automatically, no?

Reply:
According to mdn, in js "Any non-string value specified is converted automatically into a string." But typescript doesn't like implicit conversions, and yells if you try to pass a non-string to `setAttribute`.

I think the readability here might be a case of different brains are different, because it also took me a bit to figure out how checking `currentValue === "false"` was equivalent to toggling the value. (I was thinking of it in the same way as toggling any two strings, i.e. `x === "bagel" ? "orange" : "bagel"`). We could take an informal poll in fandom coders :P

Reply:
Eh, we can see if anyone goes to change it with time :P I'm ok relenting.